lib/ts/recipe/session/sessionAuth.tsx

```diff
@@ -16,12 +16,14 @@
 /*
  * Imports.
  */
-import React, { useEffect, useState, useContext, useCallback, useRef } from "react";
+import React, { useEffect, useState, useContext, useRef } from "react";
 import SessionContext, { isDefaultContext } from "./sessionContext";
 import Session from "./recipe";
-import { RecipeEvent, SessionContextType } from "./types";
+import { RecipeEventWithSessionContext, SessionContextType } from "./types";
 import { doesSessionExist, getJWTPayloadSecurely, getUserId } from "./index";
 
+// if it's not the default context, it means SessionAuth from top has
+// given us a sessionContext.
 const hasParentProvider = (ctx: SessionContextType) => !isDefaultContext(ctx);
 
 type PropsWithoutAuth = {
```

Comment:
Props should be adjusted as well, because when `requireAuth=true` we require either `redirectToLogin` or `onSessionExpired`

Reply:
That's already done right? Why further adjustment? What benefit does it give us?

Reply:
No.. not true, cause `redirectToLogin` is meant to be used for if this component is mounted and no session exists, and `requireAuth=true`. Whilst `onSessionExpired` is for session expiry.

We use `redirectToLogin` as a fallback for session expiry in case `onSessionExpired` is not provided, which may be a little strange, but we had discussed that it was OK.